src/key_defs.h

```diff
@@ -25,6 +25,16 @@ typedef union Key_ {
   };
   uint16_t raw;
 
+  Key_() = default;
```

Comment:
Is this necessary? I thought the default constructor would be created automatically. Is that a difference between `class` and `union`?

Reply:
Yes it is. If you don't implement any constructor, the compiler synthesizes a default constructor for you. Otherwise, as in the PR, the default constructor will be missing. Then you use the `... = default;` syntax to inform the compiler to generate the default constructor anyway. 

FWIW, this can be used with other stuff that is auto-generated, like assignment operators, too. It's sibling is `... = delete;` which is used to suppress auto-generated stuff.